plugins/dev9ghzdrk/PS2Edefs.h

```diff
@@ -48,12 +48,25 @@
 
 #include "PS2Etypes.h"
 
-#ifdef __linux__
-#define CALLBACK
-#else
+#ifndef __linux__
 #include <windows.h>
 #endif
 
+#if defined(_MSC_VER)
+    //  Microsoft
+    #define EXPORT __declspec(dllexport)
+    #define IMPORT __declspec(dllimport)
+#elif defined(__GNUC__)
+    //  GCC
+    #define EXPORT __attribute__((visibility("default")))
```

Comment:
Probably should be `__attribute__((stdcall, externally_visible, visibility("default")))`. The plugin interface uses the stdcall calling convention.

It might be nice to include `extern "C"` as well in EXPORT.

Reply:
cdecl is caller clean and stdcall is callee clean, so if there's a mismatch the stack cleanup will either happen twice or not at all, and that's not going to work out well.